src/gui/screen_menu_lan_settings.cpp

```diff
@@ -24,6 +24,13 @@ ScreenMenuConnectionBase::ScreenMenuConnectionBase(uint32_t dev_id, const char *
 void ScreenMenuConnectionBase::refresh_addresses() {
     char str[ADDR_LEN];
 
+    if (const auto fw_version = espif_firmware_version() > 0) {
```

Comment:
This would set the fw_version to bool of `espif_firmware_version() > 0`, so it would always report `1` as the version.
You can do two things:
* Omit the `> 0`. The fw_version type is `uint8_t`, so the cast to bool is equivalent to it anyway.
* Make it `if (const auto fw_version = espif_firmware_version(); fw_version > 0) {`

Reply:
Indeed. Fixed.